serve/mlc_serve/model/paged_cache_model.py

```diff
@@ -147,41 +292,100 @@ def set_size(self, sequence_ids: List[SequenceId], target_sizes: List[int]):
                     if self.block_sliding_window:
                         block_idx %= self.block_sliding_window
 
-                    block_number = self.kv_cache.block_tables[id][block_idx]
+                    block_number = self.kv_cache.prompt_block_tables[id][block_idx]
                     block_offset = i % self.block_size
                     slot = block_number * self.block_size + block_offset
                     self.kv_cache.slot_mappings[id].append(slot)
 
     def get_cache(self):
         return self.kv_cache
 
-    def allocate(self, request_id: RequestId, num_tokens: int):
+    def allocate(self, request_id: RequestId, num_tokens: int, num_sequences: int):
         """
         Allocate cache space for a prefill request, raise error if there is no space.
         """
-        prompt_seq_id = SequenceId(request_id, 0)
+        prompt_seq_id = get_prompt_sequence_id(request_id)
         self.set_size([prompt_seq_id], [num_tokens])
-        self.allocated_tokens[prompt_seq_id] = num_tokens
+        self.allocated_prompt_tokens[prompt_seq_id] = num_tokens
+
+        last_block_partially_shared = num_sequences > 1 and (
+            num_tokens % self.block_size != 0
+        )
+
+        if self.sliding_window:
```

Comment:
Orthogonal to this PR, but what happens if we have more prompt tokens than sliding window size?